UserAgeCalculation

```diff
@@ -5,26 +5,48 @@ namespace UserAgeCalculation
     class Program
     {
         static void Main(string[] args)
-        {
+       {
             Console.Write("Enter your name: ");
             string userName = Console.ReadLine();
 
+            int currentYear = DateTime.Now.Year;
+            int yearOfBirth = 0;
 
             Console.Write("Enter your year of birth: ");
-            int yearOfBirth = Convert.ToInt32(Console.ReadLine());
+            string yearOfBirthString = Console.ReadLine();
 
-            int currentYear = DateTime.Now.Year;
-            int userAge = currentYear - yearOfBirth; 
+            while (true)
+            {
+                if (!int.TryParse(yearOfBirthString, out yearOfBirth))
+                {
+                    Console.WriteLine("Invalid input string, year of birth should be a four digit number");
+                    Console.Write("Enter your year of birth: ");
+                    yearOfBirthString = Console.ReadLine();
+                    continue;
+                }
 
+                if ((currentYear - 100) <= yearOfBirth && yearOfBirth < currentYear)
+                    break;
+                {
```

Comment:
this braces are unnecessary. you have to use it only if your condition's action contains more than one line. For example if above contain only one action if condition met. In your case this braces are unnecessary because there's no condition before it. Please remove it.